helios.js

```diff
@@ -28652,6 +28709,98 @@ class InlineDatum extends Datum {
 	}
 }
 
+/**
+ * Encoding for Transcation Matadatum
+ */
+export class MetadataString extends CborData {
+	/** @type {number[]} */
+	#bytes;
+
+	/**
+	 * @param {number[]} bytes
+	 */
+	constructor(bytes) {
+		super();
+		this.#bytes = bytes;
+	}
+
+	/**
+	 * Applies utf-8 encoding
+	 * @param {string} s
+	 * @returns {MetadataString}
+	 */
+	static fromString(s) {
+		let bytes = stringToBytes(s);
+
+		return new MetadataString(bytes);
+	}
+
+	get bytes() {
+		return this.#bytes.slice();
+	}
+
+	/**
+	 * @returns {string}
+	 */
+	toHex() {
+		return bytesToHex(this.#bytes);
+	}
+
+	toString() {
+		return `#${this.toHex()}`;
+	}
+
+	/**
+	 * @returns {number[]}
+	 */
+	toCbor() {
```

Comment:
If we would use javascript types instead of `MetadataString`: I would move this method to `TxMetadata` as a static method.